src/lib/dfp/prepare-googletag.ts

```diff
@@ -94,7 +92,7 @@ export const init = (): Promise<void> => {
 			let canRun = true;
 
 			if (consentState.canTarget) {
-				window.googletag.cmd.push(setPublisherProvidedId);
+				window.googletag.cmd.push(() => setPublisherProvidedId);
```

Comment:
`setPublisherProvidedId` won't be called here!

Also even if it was the promise won't be waited on before execution here (I don't think), so there could be a chance that the id isn't set before googletag requests are made. 

Maybe something like:
```js
await new Promise((resolve) => {
	window.googletag.cmd.push(() => setPublisherProvidedId().then(resolve));
});
```

Or there may be a better solution